calculator.js

```diff
@@ -1,5 +1,140 @@
-const exampleAdditionInput = {
+// Set array of valid operations 
+const validOperations = ['+', 'add', '-', 'subtract', '/', 'divide', '*', 'multiply']
+
+// Helper functions
+function validateNums(input) {
+  if (isNaN(input.num1) || isNaN(input.num2)) {
+    return false; 
+  } else {
+    return true; }
+}
+
+function validateOperator(input) {
+  if (!validOperations.includes(input.operation)) {
+    return false; }
+  else {
+    return true; }
+}
+
+const addMe = function(num1, num2) {
+  console.log(num1, num2);
+  return num1 + num2;
+}
+
+const subMe = function(num1, num2) {
+  return num1 - num2;
+}
+
+const divMe = function(num1, num2) {
+  return num1 / num2;
+}
+
+const mulMe = function(num1, num2) {
+  return num1 * num2;
+}
+
+// Main function 
+function calculate(input) {
+  if (!validateNums(input)) {
+    console.log('I spot an error. Looks like one of your values was not numeric.');
+    return NaN;
+  } else if (!validateOperator(input)) {
+    console.log('I spot an error. Looks like your operator isn\'t for addition, subtraction, multiplication, or division.'); 
+    return NaN;
+  }
+
+  const num1 = input.num1;
+  const num2 = input.num2;
+  const operator = input.operation;
+
+  if (operator === '+' || operator === 'add') {
+    return addMe(num1, num2);
```

Comment:
I like the decision to return a value from this function instead of printing it directly - that makes this code easier to test, and would make it easier to extend in the future.